src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -1927,36 +1927,52 @@ public DynamicObject sortVeryShort(VirtualFrame frame, DynamicObject array, NotP
             return createArray(store.getArray(), size);
         }
 
-        @Specialization(guards = { "!isEmptyArray(array)", "!isSmall(array)", "isIntArray(array)" },
+        @Specialization(guards = { "forInt(strategy)", "!isEmptyArray(array)", "!isSmall(array)", "strategy.matches(array)" },
```

Comment:
Can you add the forInt() & friends as the last guards?
So it's clearer the specializations are complementary re isEmptyArray/isSmall.

I think `isIntStrategy(strategy)` would be clearer, too.